packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -113,6 +117,29 @@ export abstract class InputMediaDeviceManager<
     }
   }
 
+  /**
+   * Registers a filter that will be applied to the stream.
+   *
+   * The registered filter will get the existing stream, and it should return
+   * a new stream with the applied filter.
+   *
+   * @param filter the filter to register.
+   */
+  async registerFilter(filter: MediaStreamFilter) {
+    this.filters.push(filter);
+    if (this.state.status === 'enabled') {
+      await this.disable();
+      await this.enable();
+    }
+    return async () => {
+      this.filters = this.filters.filter((f) => f !== filter);
+      if (this.state.status === 'enabled') {
+        await this.disable();
```

Comment:
We have the `applySettingsToStream` method to avoid having to do a status update (`enabled` -> `disabled` -> `enabled`) on settings change, can't we use that method here as well?

Reply:
Good point. I'll adjust accordingly!

Reply:
Adjusted in [cef2749](https://github.com/GetStream/stream-video-js/pull/1283/commits/cef27495475061e20fc939bf010b6dbfe5cfdf0b)